### ISSUE TITLE
Data Point Event Notify Work Item - Running after timer is shutdown

### ISSUE THREAD

**Issue by terrypacker** [infiniteautomation/ma-core-public#1727]
This was seen during shutdown with a large queue of medium priority events:

```
ERROR 2021-05-11T22:00:23,930 (com.serotonin.m2m2.rt.maint.BackgroundProcessingImpl$RejectableWorkItemRunnable.run:512) - Error in work item
java.lang.IllegalStateException: Timer already cancelled.
	at com.serotonin.timer.RealTimeTimer.scheduleImpl(RealTimeTimer.java:111) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.timer.AbstractTimer.schedule(AbstractTimer.java:27) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.rt.maint.BackgroundProcessingImpl.schedule(BackgroundProcessingImpl.java:72) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.util.timeout.TimeoutTask.<init>(TimeoutTask.java:41) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.util.timeout.TimeoutTask.<init>(TimeoutTask.java:31) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.infiniteautomation.xxx.dem.salesforce.SalesforceCaseEventHandlerRT.eventRaised(SalesforceCaseEventHandlerRT.java:131) ~[?:?]
	at com.serotonin.m2m2.rt.EventManagerImpl.handleRaiseEvent(EventManagerImpl.java:999) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.rt.EventManagerImpl.raiseEvent(EventManagerImpl.java:260) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.rt.event.detectors.PointEventDetectorRT.raiseEvent(PointEventDetectorRT.java:45) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.rt.event.detectors.StateDetectorRT.setEventActive(StateDetectorRT.java:109) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.rt.event.detectors.TimeDelayedEventDetectorRT.scheduleJob(TimeDelayedEventDetectorRT.java:34) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.rt.event.detectors.StateDetectorRT.changeStateActive(StateDetectorRT.java:69) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.rt.event.detectors.StateDetectorRT.pointChanged(StateDetectorRT.java:82) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.rt.DataPointEventNotifyWorkItem.execute(DataPointEventNotifyWorkItem.java:70) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.m2m2.rt.maint.BackgroundProcessingImpl$RejectableWorkItemRunnable.run(BackgroundProcessingImpl.java:509) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.timer.Task.runTask(Task.java:183) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.timer.TaskWrapper.run(TaskWrapper.java:23) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at com.serotonin.timer.OrderedThreadPoolExecutor$OrderedTaskCollection.run(OrderedThreadPoolExecutor.java:313) ~[mango-4.0.0-SNAPSHOT.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by terrypacker** [comment]
@jazdw can you take a look at the commit here and close if happy.

**Comment by jazdw** [comment]
Pretty hard to follow without an explanation, call me if you feel the need to discuss.
